scripts/rebalance-queue-masters

```diff
@@ -74,11 +74,40 @@ set_temp_queue_policies()
     local updated_master=''
     local policy_set_master=''
     local -i count=0
-    local -i master_idx=0
 
     IFS=$'\n'
-    for queue_and_pid in $(rabbitmqctl -q -p "$vhost" list_queues name pid | grep -E "$queue_regex")
-    do
+    while true; do
+        # List of nodes
+        node_list=$(rabbitmqctl eval 'rabbit_nodes:all_running().' | sed -e "s/[][']//g" | tr ',' '\n')
+
+        # List of masters extracted from node_list + list_queues masters.
+        nodes_and_masters=$((echo "$node_list"; rabbitmqctl -q -p "$vhost" list_queues pid) | grep rabbit | sed -e 's/[><]//g' -e 's/\(\.[[:digit:]][[:digit:]]*\)\{3\}//g' | sort | uniq -c | sort -n)
+
+        # Number of nodes in cluster
+        num_nodes=$(echo "$nodes_and_masters" | wc -l)
```

Comment:
Isn't this the number of items in `node_list`?

Reply:
if you have a node with 0 queue master (after node reboot, for example), this node isn't in list_queues list, for that you need add `node_list` variable to `nodes_and_masters`. So, `num_nodes` depends on `nodes_and_master`, and `nodes_and_master` depends on `node_list`. The rest of `max_*` and `min_*` variables depend on `nodes_and_masters` Probably,  could be use `num_nodes=$(echo $node_list" | wc -l)` but I feel confortable, and it's probably clearer and easier to undestan